t/sanity.t

```diff
@@ -468,3 +468,34 @@ lock 2: unlock: nil, unlocked
 --- no_error_log
 [error]
 
+
+
+=== TEST 14: serial lock and unlock with safe_add
+--- http_config eval: $::HttpConfig
+--- config
+    location = /t {
+        content_by_lua '
+            local lock = require "resty.lock"
+            for i = 1, 2 do
+                local lock = lock:new("cache_locks", {safe_add = true})
+                local elapsed, err = lock:lock("foo")
+                ngx.say("lock: ", elapsed, ", ", err)
+                local ok, err = lock:unlock()
+                if not ok then
+                    ngx.say("failed to unlock: ", err)
+                end
+                ngx.say("unlock: ", ok)
+            end
+        ';
+    }
+--- request
+GET /t
+--- response_body
+lock: 0, nil
+unlock: 1
+lock: 0, nil
+unlock: 1
```

Comment:
added the new test case . ^_^

On Sun, Dec 6, 2015 at 12:45 PM, Yichun Zhang <notifications@github.com>
wrote:

> In t/sanity.t
> <https://github.com/openresty/lua-resty-lock/pull/6#discussion_r46766560>:
>
> > +                ngx.say("lock: ", elapsed, ", ", err)
> > +                local ok, err = lock:unlock()
> > +                if not ok then
> > +                    ngx.say("failed to unlock: ", err)
> > +                end
> > +                ngx.say("unlock: ", ok)
> > +            end
> > +        ';
> > +    }
> > +--- request
> > +GET /t
> > +--- response_body
> > +lock: 0, nil
> > +unlock: 1
> > +lock: 0, nil
> > +unlock: 1
>
> Need to test the case that safe_add actually returned the "no memory"
> error.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openresty/lua-resty-lock/pull/6/files#r46766560>.
>



-- 

*MembPhis*
My github: https://github.com/membphis
Our Book: OpenResty Best Practices
<https://www.gitbook.com/book/moonbingbing/openresty-best-practices>